lib/util.js

```diff
@@ -62,17 +62,26 @@ exports.deprecate = function(fn, msg) {
   }
 
   var warned = false;
+  const warnedLines = new Set();
+
   function deprecated() {
+    var location = (new Error()).stack.match(/.*?\n.*\n.*?\((.*?:\d+):/)[1];
```

Comment:
That line above was produced by just `console.log(stack.match(/.*?\n.*\n.*?\((.*?:\d+):/)[1]);`. I think the lines without parens are anonymous functions.